src/Nio/MidiIn.cpp

```diff
@@ -147,7 +170,96 @@ void MidiIn::midiProcess(unsigned char head,
             ev.value   = (num + value * (int) 128) - 8192;
             InMgr::getInstance().putEvent(ev);
             break;
+        case 0x79: // reset to power-on state
+            ev.type    = M_CONTROLLER;
+            ev.channel = chan;
+            ev.num     = C_resetallcontrollers;
+            ev.value   = 0;
+            InMgr::getInstance().putEvent(ev);
+                break;
+    }
+    
+    /* MTC handling */
+    int subtype = (value >> 4) & 0xf;
+    int val  = (value & 0xf);
+    ev.channel = 0;
+
+    switch(head) {
+        case 0xF1: /* M_TC - MIDI Time Code Quarter Frame */
+            switch (subtype)
+            {
+            case 0:
+            {
+                frameLo = val;
+                break;
+            }
+            case 1:
+            {
+                frame = frameLo + (val * 16);
+                break;
+            }
+            case 2:
+            {
+                secondLo = val;
+                break;
+            }
+            case 3:
+            {
+                second = secondLo + (val * 16);
+                break;
+            }
+            case 4:
+            {
+                minuteLo = val;
+                break;
+            }
+            case 5:
+            {
+                minute = minuteLo + (val * 16);
+                break;
+            }
+            case 6:
+            {
+                hourLo = (val & 31);
+                break;
+            }
+            case 7:
+            {
+                hour = hourLo + ((val * 16) & 31);
+                if(frame==28) {
+                    ev.type  = M_TC;
+                    ev.num = 0;
+                    ev.value = second+60*minute+3600*hour;
+                    InMgr::getInstance().putEvent(ev);
+                }
+                break;
+            }
+            }
+            break;
+
+        case 0xF2: /* M_SPP - Song Position Pointer*/
+            ev.type  = M_SPP;
+            ev.value = num + 128 * value;
+            InMgr::getInstance().putEvent(ev);
+            break;
+        case 0xF8: /* M_CLOCK */
+            ev.type  = M_CLOCK;
+            InMgr::getInstance().putEvent(ev);
+            break;
+        case 0xFA: /* M_START */
+            ev.type  = M_START;
+            InMgr::getInstance().putEvent(ev);
+            break;
+        case 0xFB: /* M_STOP */
```

Comment:
small note here, 0xFB is continue, and 0xFC is stop.
you have it in reverse.

Reply:
> small note here, 0xFB is continue, and 0xFC is stop.
> you have it in reverse.

thx